### ISSUE TITLE
Primus ignores my http server

### ISSUE THREAD

**Issue by omidh28** [primus/primus#544]
**Version:** 6.0.9

**Environment:**
  - **Operating system:** Ubuntu
  - **Browser:** Chrome
  - **Node.js:** 7.4.0

I use Primus for building real-time game servers. A few users reported sudden interruptions and disconnections which I didn't see before however a few days ago when I was playing and I got disconnected, I checked logs and it was CORS problem! 

I'm not very familiar with CORS but why it happens sometimes and not all the time? I've switched from engine.io to uws so it might fix the problem that I couldn't reproduce.

After switching to uws I noticed there isn't any CORS header in server's response.

These are the headers response from server:

> Connection: Upgrade
> Sec-WebSocket-Accept:h4UMlOFEGteuQ8YdQkUpYnsT+JI=
> Sec-WebSocket-Version:13
> Upgrade:websocket
> WebSocket-Server:uWebSockets

and in another request: 

> accept-ranges:bytes
> cache-control:public, max-age=0
> Connection:close
> date:Wed, 08 Feb 2017 14:42:50 GMT
> etag:W/"15b78-15a1dfd125f"
> last-modified:Wed, 08 Feb 2017 13:50:08 GMT
> x-powered-by:Express

note that **x-powered-by:Express**, I don't even have express js installed and I use simple node http server:

``` Javascript
const httpServer = require('http').createServer((req, res) => {});
const primus = new Primus(httpServer, { parser: 'JSON', transformer: 'uws' });
 

// primus.on(...

httpServer.listen(8000, () => {
    console.log('running socket server under port 8000');
});

```

I tried `res.setHeader...` in my http server but it seems like everything I write in http server configuration are ignored even though it is still run under the port (8000) I told it to run.

In summary These are my questions and confusions:

1. Why does Primus ignore my http server and use its own express version?
2. Why uws does not include CORS headers but still works cross-origin?

**Comment by lpinca** [comment]
> 1. Why does Primus ignore my http server and use its own express version?

Primus doesn't use its own server, in fact, you have to explicitly provide your own HTTP server in the constructor and we don't use the `x-powered-by` header in any part of our codebase.

> 2. Why uws does not include CORS headers but still works cross-origin?

Because the same-origin policy does not apply to WebSocket.

**Comment by omidh28** [comment]
@lpinca so you are saying that I don't need to care about CORS even for socket handshaking?

**Comment by lpinca** [comment]
Yes, if you use a WebSocket only transformer (websockets, faye, uws) you will not have CORS issues with the primus server.
 
> Also is it okay that I left my HTTP server body function empty?

That request handler is only run for requests not captured by primus (not made by the primus client) so it doesn't matter unless you want to handle them.
